ts/nni_manager/common/globals/log_stream.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+/**
+ *  The underlying IO stream of loggers.
+ *
+ *  Normal modules should not use this directly. Use "common/log.ts" instead.
+ **/
+
+import fs from 'fs';
+import { setTimeout } from 'timers/promises';
+import util from 'util';
+
+import type { NniManagerArgs } from './arguments';
+import type { NniPaths } from './paths';
+
+export interface LogStream {
+    writeLine(line: string): void;
+    close(): Promise<void>;
+}
+
+const writePromise = util.promisify(fs.write);
+
+class LogStreamImpl implements LogStream {
+    private buffer: string[] = [];
+    private flushing: boolean = false;
+    private logFileFd: number;
+    private toConsole: boolean;
+
+    constructor(logFile: string, toConsole: boolean) {
+        this.logFileFd = fs.openSync(logFile, 'a');
+        this.toConsole = toConsole;
+    }
+
+    public writeLine(line: string): void {
+        this.buffer.push(line);
+        this.flush();
```

Comment:
How do you catch errors raised from flush?

Reply:
Not handled yet.
Seems we have discussed it in the review meeting. But I forgot to fix it.

Reply:
Done. I decided to use synchronous write for robust logger. In critical area speed is not a thing.